src/utils.jl

```diff
@@ -77,6 +77,55 @@ function sideof(p::Point{2,T}, c::Chain{2,T}) where {T}
   ifelse(isapprox(w, zero(T), atol=atol(T)), :OUTSIDE, :INSIDE)
 end
 
+"""
+    sideof(point, mesh)
+
+Determines whether a `point` is inside, outside or on the surface of a `mesh`.
+Possible results are `:INSIDE`, `:OUTSIDE`, or `:ON`.
+
+### Notes
+
+Uses a ray-casting algorithm.
+"""
+function sideof(point::Point{3,T}, mesh::Mesh{3,T}) where {T}
+  if !(eltype(mesh) <: Triangle)
+    throw(ArgumentError(
+          "This function only works for surface meshes with triangles as elements."))
+  end
+
+  z = last.(coordinates.(extrema(mesh)))
+  r = Ray(point, Vec(zero(T), zero(T), 2 * (z[2] - z[1])))
+
+  intersects = false
+  edgecrosses = 0
+  ps = Point{3,T}[]
+  for e in mesh
+    result = intersection(r, e) do I
+      if type(I) == CrossingRayTriangle
+        intersects = !intersects
+      elseif type(I) ∈ (EdgeTouchingRayTriangle,
+                        CornerTouchingRayTriangle,
+                        TouchingRayTriangle)
+        return :ON
+      elseif type(I) == EdgeCrossingRayTriangle
+        edgecrosses += 1
+      elseif type(I) == CornerCrossingRayTriangle
+        p = get(I)
+        if !any(==(p), ps)
+            push!(ps, p)
```

Comment:
@kylebeggs should ps be a Set? This pattern of checking if the point is in the list and then pushing it to the list should be natural with the Set data structure, right? You can always push to the Set and it will contain unique members all the time.

Reply:
Yes that makes sense.

Reply:
For some reason `Set` gives the wrong answer here
```
Point(0.25, 0.25, 0.9999999999999999)
Set{Point3} with 1 element:
  Point(0.25, 0.25, 1.0)
p ∉ ps = true
```
 but with a vector we get the correct result:
```
Point(0.25, 0.25, 0.9999999999999999)
1-element Vector{Point3}:
 Point(0.25, 0.25, 1.0)
p ∉ ps = false
```

Any ideas? I thought both of a `Set` and a `Vector` should compare with the same tolerance? This example probably demonstrates we should just use `isapprox` anyways.

Reply:
Maybe this is due to the difference between == and isequal. I think Set uses isequal, and we probably need to implement it for Point as well.

Reply:
== and isequal just treat NaN, missing, and 0.0 differently. I don't think it should matter in the example above? i know its not ideal, but we can use the vector for now?

Reply:
Sounds good, we just need to replace == by \isapprox. See https://github.com/JuliaGeometry/Meshes.jl/pull/384

We will refactor some basic concepts in the near future.